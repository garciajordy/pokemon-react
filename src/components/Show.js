import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

function Show({
  pokemonName, pokeImg, style, weight, height,
}) {
  return (
    <div className="show">
      <a href="/">Back</a>
      <div className="poke">
        <h2>{pokemonName}</h2>
        <img src={pokeImg} alt={pokemonName} />
        <h1>
          {pokemonName}
        </h1>
        <p>
          <span>{'Type: '}</span>
          {style}
        </p>
        <p>
          <span>
            {'Height: '}
          </span>
          {height / 10}
          {' M'}
        </p>
        <p>
          <span>
            {'Weight: '}
          </span>
          {weight / 10}
          {' KG'}
        </p>
      </div>

    </div>
  );
}

Show.propTypes = {
  pokemonName: PropTypes.string,
  pokeImg: PropTypes.string,
  weight: PropTypes.string,
  height: PropTypes.string,
  style: PropTypes.string,
};
Show.defaultProps = {
  pokemonName: 'test',
  pokeImg: 'test',
  weight: 'test',
  height: 'test',
  style: 'test',
};
const mapStateToProps = (state) => ({
  pokemonName: state.searchReducer.name,
  pokeImg: state.searchReducer.img,
  weight: state.searchReducer.weight,
  height: state.searchReducer.height,
  style: state.searchReducer.style,
});

export default connect(mapStateToProps)(Show);
