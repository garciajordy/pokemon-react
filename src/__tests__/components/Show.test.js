import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Show from '../../components/Show';
import store from '../../store/index';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><Show /></Provider>);
  expect(asFragment(<Provider store={store}><Show /></Provider>)).toMatchSnapshot();
});

it('displays heading', () => {
  render(<Provider store={store}><Show /></Provider>);
  expect(screen.getByText('Back')).toBeTruthy();
});
