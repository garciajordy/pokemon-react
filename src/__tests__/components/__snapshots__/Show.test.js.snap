// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="show"
  >
    <a
      href="/"
    >
      Back
    </a>
    <div
      class="poke"
    >
      <h2>
        test
      </h2>
      <img
        alt="test"
        src="test"
      />
      <h1>
        test
      </h1>
      <p>
        <span>
          Type: 
        </span>
        test
      </p>
      <p>
        <span>
          Height: 
        </span>
        NaN M
      </p>
      <p>
        <span>
          Weight: 
        </span>
        NaN KG
      </p>
    </div>
  </div>
</DocumentFragment>
`;
